__tests__/app.test.js

```diff
@@ -162,6 +161,124 @@ describe("POST /api/article/:aritcle_id/comments", () => {
         username: "jessjelly",
         body: "Great article!",
       })
-      .expect(404);
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Incorrect article ID");
+      });
+  });
+  test("should return 400 error if passed an invalid article ID", () => {
+    return request(app)
+      .post("/api/articles/potato/comments")
+      .send({
+        username: "jessjelly",
+        body: "Great article!",
+      })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Invalid article ID");
+      });
+  });
+  test(`should return 404 error if passed a username that doesn't exist`, () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "MichaelScott24",
+        body: "Great article!",
+      })
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Username not found");
+      });
+  });
+  test("should return 400 error if no body is passed", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "jessjelly",
+      })
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("No body property on request");
+      });
+  });
+});
+
```

Comment:
Excellent! Just one more test needed:
404- Username not found (i.e. include a username in the req.body that doesn't exist on the users table. This can be handled using PSQL error code.